src/Channels/CommonCache.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System.Numerics;
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+
+namespace Channels
+{
+    // Move to text library?
```

Comment:
Still not happy with how it behaves/the asm it generates; its better - but not good. Will follow up with the issues with the splat/broadcast .ctor in coreclr and see if we can make that better.